### PR TITLE
More types support for QgsVectorFileWriter

### PR DESCRIPTION
**Pull request by m-kuhn**
In particular but not limited to Geometry Collections

### CODE REVIEW

File: src/core/qgsvectorfilewriter.cpp [hunk 1]
```diff
@@ -1551,6 +1565,10 @@ bool QgsVectorFileWriter::driverMetadata( const QString& driverName, QgsVectorFi
   return false;
 }
 
+OGRwkbGeometryType QgsVectorFileWriter::ogrTypeFromWkbType( QgsWKBTypes::Type type )
+{
+  return static_cast<OGRwkbGeometryType>( type );
```
Comment (nyalldawson) [inline src/core/qgsvectorfilewriter.cpp]: should this method also handle converting Z types to 25D types?

Comment (m-kuhn) [inline src/core/qgsvectorfilewriter.cpp]: I don't know, but things like this was why I introduced this method instead of using straight casting.

Comment (nyalldawson) [inline src/core/qgsvectorfilewriter.cpp]: I'm not 100% sure, but I think it does need to convert. See http://www.gdal.org/ogr__core_8h.html#a800236a0d460ef66e687b7b65610f12a for the supported types.

Comment (m-kuhn) [inline src/core/qgsvectorfilewriter.cpp]: What should we do to M and ZM types?

Comment (m-kuhn) [inline src/core/qgsvectorfilewriter.cpp]: And special ones like 

      CircularStringZ = 1008,
      CompoundCurveZ = 1009,
      CurvePolygonZ = 1010,
      MultiCurveZ = 1011,
      MultiSurfaceZ = 1012,

Comment (nyalldawson) [inline src/core/qgsvectorfilewriter.cpp]: Measure values aren't supported by gdal - they are converted by some providers to z values (See http://www.gdal.org/drv_shapefile.html). We could either do the same or drop them. CircularStringZ/... curved geometries with z are supported in gdal >= 2.0

File: src/core/geometry/qgswkbtypes.cpp [hunk 2]
```diff
@@ -236,6 +236,16 @@ QgsWKBTypes::Type QgsWKBTypes::dropM( QgsWKBTypes::Type type )
   return returnType;
 }
 
+QgsWKBTypes::Type QgsWKBTypes::to25D( QgsWKBTypes::Type type )
```
Comment (nyalldawson) [inline src/core/geometry/qgswkbtypes.cpp]: this is a protected class - you'll need to add unit tests for this!

Comment (m-kuhn) [inline src/core/geometry/qgswkbtypes.cpp]: done

**Comment by nyalldawson** [comment]
Ok - LGTM. Hopefully this also fixes an issue I was running into with adding z dimension to a 2d layer. I'll test when this is merged :)

**Comment by m-kuhn** [comment]
Merged in e351d24a7d5cc205b41834d26b673a5204a69bd8

**Pull request closed without merging**